src/languageservice/utils/json.ts

```diff
@@ -50,18 +50,23 @@ export function stringifyObject(
       let result = (depth === 0 && settings.newLineFirst) || depth > 0 ? '\n' : '';
       for (let i = 0; i < keys.length; i++) {
         const key = keys[i];
+        const isObject = typeof obj[key] === 'object';
+        const colonDelimiter = isObject ? ':' : ': '; // add space only when value is primitive
+        const parentArrayCompensation = isObject && /^\s|-/.test(key) ? '  ' : ''; // add extra space if parent is an array
```

Comment:
extra space should be calculated based on the indentation

Reply:
Thanks, it makes sense
I fixed it